spark/src/main/scala/org/apache/comet/DataTypeSupport.scala

```diff
@@ -37,8 +39,7 @@ trait DataTypeSupport {
 
   private def isGloballySupported(dt: DataType): Boolean = dt match {
     case ByteType | ShortType
-        if CometSparkSessionExtensions.usingDataFusionParquetExec(SQLConf.get) &&
-          !CometConf.COMET_SCAN_ALLOW_INCOMPATIBLE.get() =>
+        if usingParquetExecWithIncompatTypes(CometConf.COMET_NATIVE_SCAN_IMPL.get(SQLConf.get)) =>
```

Comment:
can we create a global lazy val to reduce number of `CometConf.COMET_NATIVE_SCAN_IMPL.get(SQLConf.get)` ?

Reply:
fair enough, it can be a function.